x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/storage/ReactiveStorageDeciderService.java

```diff
@@ -101,14 +102,25 @@ public List<DiscoveryNodeRole> roles() {
 
     @Override
     public AutoscalingDeciderResult scale(Settings configuration, AutoscalingDeciderContext context) {
+        AllocationState allocationState = allocationState(context);
+        var assignedBytesUnmovableShards = allocationState.storagePreventsRemainOrMove0();
+        var unassignedBytesUnassignedShards = allocationState.storagePreventsAllocation0();
         AutoscalingCapacity autoscalingCapacity = context.currentCapacity();
         if (autoscalingCapacity == null || autoscalingCapacity.total().storage() == null) {
-            return new AutoscalingDeciderResult(null, new ReactiveReason("current capacity not available", -1, -1));
+            return new AutoscalingDeciderResult(
+                null,
+                new ReactiveReason(
+                    "current capacity not available",
+                    -1,
+                    -1,
+                    unassignedBytesUnassignedShards.unassignedShards(),
```

Comment:
I'm wondering if it would be better to return an empty list in this case as we're signalling that we cannot make an autoscaling decision in the current state.